Mathlib/Topology/Bases.lean

```diff
@@ -671,10 +671,38 @@ class _root_.FirstCountableTopology : Prop where
 
 attribute [instance] FirstCountableTopology.nhds_generated_countable
 
-namespace FirstCountableTopology
+/-- If `β` is a first-countable space, then its induced topology via `f` on `α` is also
+first-countable. -/
+theorem firstCountableTopology_induced (α β : Type*) [t : TopologicalSpace β]
+    [FirstCountableTopology β] (f : α → β) : @FirstCountableTopology α (t.induced f) := by
+  letI := t.induced f
+  refine ⟨fun a ↦ ?_⟩
+  have nhds_fa := @FirstCountableTopology.nhds_generated_countable β _ _ (f a)
+  rw [Filter.isCountablyGenerated_iff_exists_antitone_basis] at *
+  obtain ⟨B, hB⟩ := nhds_fa
+  refine ⟨fun n ↦ preimage f (B n), ⟨?_, fun _ _ le ↦ preimage_mono <| hB.2 le⟩⟩
+  rw [(@induced_iff_nhds_eq β α _ _ f).mp rfl a]
+  apply hB.toHasBasis.comap
```

Comment:
Oh, wow... Where did all those lines go? :golf: 

The `▸` is slick. I guess the `;` is not necessary, however.

Thanks a lot!

Reply:
The `▸` is really not the key part (it's a bit of a drug, I just have to try if it works instead of `rw`...). The key part is that `inferInstance` (and really [Filter.comap.isCountablyGenerated](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Bases.html#Filter.comap.isCountablyGenerated) is doing all of the work!

And indeed the `;` is not necessary, I can never remember the syntax in term mode...